### PR TITLE
UI Component/#72 - WinModal

### LINKED ISSUE #72: UI Component/#72 - Win Modal

### PR DESCRIPTION
**Pull request by Enessetere**
closes #72

### CODE REVIEW

File: src/app/components/WinModal.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    );
+  }
+
+  private createImage({ theName, theAvatar }: Player): HTMLElement {
+    return Render.elementFactory('img', {
+      src: theAvatar,
+      alt: `${theName}-avatar`,
+    });
+  }
+}
```
Comment (synowa) [inline src/app/components/WinModal.ts:28]: As the red sign is suggesting, newline at the end of this file could be added.

**Review by synowa — APPROVED** [review]

File: src/app/components/WinModal.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
+import { Player } from '~src/Player';
```
Comment (vieraboschkova) [inline src/app/components/WinModal.ts:1]: should you remove tildas from paths?

File: src/app/components/WinModal.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+export class WinModal extends ModalBasic {
+  create(player: Player): void {
+    this.renderBasicModal(
+      'CONGRATULATIONS!',
+      `${player.theName} has won!`,
+      this.createImage(player),
+    ) as WinModal;
+  }
+
+  addButton(): void {
```
Comment (vieraboschkova) [inline src/app/components/WinModal.ts:14]: Perhaps addEndGameButton? It has the text 'End game - back to menu' inside, so perhaps the name of the method could be clearer. Consider also adding method definition.

File: src/app/components/WinModal.ts [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    const handleEnd = () => Render.removeElement('.modal');
+    this.createAndAppendButtonsRow(
+      'End game - back to menu',
+      handleEnd,
+    );
+  }
+
+  private createImage({ theName, theAvatar }: Player): HTMLElement {
```
Comment (vieraboschkova) [inline src/app/components/WinModal.ts:22]: Great, now i understand why name and avatar didn't work :D 🥇 It all makes sense now 🦅

File: src/app/components/WinModal.ts [hunk 5]
```diff
@@ -0,0 +1,28 @@
+import { Player } from '~src/Player';
+import { Render } from '../utils/Render';
+import { ModalBasic } from './ModalBasic';
+
+export class WinModal extends ModalBasic {
+  create(player: Player): void {
```
Comment (vieraboschkova) [inline src/app/components/WinModal.ts:6]: Consider adding method definition

**Review by vieraboschkova — APPROVED** [review]
Check if you need to remove tildas from paths, consider name changing for addButton method and add method's definitions.
Excellent destructuring! :)

**Pull request merged**